### PR TITLE
Fix ubuntu-22.04 workflow

### PR DESCRIPTION
**Pull request by CarlosEduR**

### CODE REVIEW

**Comment by CarlosEduR** [comment]
Just debugging this, seems we are having an issue with the ubuntu-22.04 workflow

```
[49/86] Building CXX object benchmarks/CMakeFiles/benchdata.dir/bench.cpp.o
[50/86] Building CXX object benchmarks/CMakeFiles/bbc_bench.dir/bbc_bench.cpp.o
[51/86] Building CXX object benchmarks/CMakeFiles/percent_encode.dir/percent_encode.cpp.o
[52/86] Linking CXX executable benchmarks/percent_encode
FAILED: benchmarks/percent_encode 
: && /usr/bin/g++-12 -O3 -DNDEBUG  benchmarks/CMakeFiles/percent_encode.dir/percent_encode.cpp.o -o benchmarks/percent_encode  src/libada.a  _deps/benchmark-build/src/libbenchmark.a  -lservo-url  /usr/lib/x86_64-linux-gnu/librt.a && :
/usr/bin/ld: cannot find -lservo-url: No such file or directory
collect2: error: ld returned 1 exit status
[[53](https://github.com/ada-url/ada/actions/runs/9735521594/job/26864830332?pr=691#step:5:54)/86] Building CXX object benchmarks/CMakeFiles/model_bench.dir/model_bench.cpp.o
ninja: build stopped: subcommand failed.
```

**Comment by anonrig** [comment]
@lemire can you take a look at this?

File: .github/workflows/ubuntu.yml [hunk 1]
```diff
@@ -28,16 +28,25 @@ jobs:
         shared: [ON, OFF]
         cxx: [g++-12, clang++-14]
     steps:
-      - uses: actions/checkout@a5ac7e51b41094c92402da3b24376905380afc29 # v4.1.6
-      - name: Setup Ninja
-        run: sudo apt-get install ninja-build
+      - uses: actions/checkout@v4.1.6
```
Comment (anonrig) [inline .github/workflows/ubuntu.yml:31]: Can you revert this line change? We use hashes due to security

**Review by anonrig — APPROVED** [review]

**Comment by CarlosEduR** [comment]
Not sure if this is the best solution, and also not sure the impact, but seems related to rust, benchmarks and servo-url. 👀

**Pull request merged**